tests/end2end/test_simple.py

```diff
@@ -429,6 +429,34 @@ def test_successful_user(self):
         upload = self.session.put(url, data=success, headers={'Script-Name': '/foo'})
         assert upload.status_code == 200
 
+    def test_successful_user_not_allowed_to_restricted_paths (self):
+        self.session.auth = ('username-pass-no-restricted-paths', 'fake-password')
+
+        # Mimic a download
+        url = '{artifacts_url}/download/{container}/'.format(
+            artifacts_url=self.artifacts_url,
+            container=self.container
+        )
+        upload = self.session.get(url)
+        assert upload.status_code == 404
```

Comment:
This is used to validate that a user with no access to restricted paths still has access to other paths.